### DISCUSSION TITLE
Calling `update()` on `QtInteractor`

### DISCUSSION THREAD

**Discussion by adam-grant-hendry** [pyvista/pyvistaqt#222] [Q&A]
When `update()` is called on `QtInteractor`, which method gets called? `BasePlotter` has an `update()` method, but if `QVTKRenderWindowInteractor` inherits from `QWidget`, it does too.

**Comment by larsoner** [comment]
Based on the standard MRO and subclass order I think it would go to the VTKRenderWindowInteractor.update() https://github.com/pyvista/pyvistaqt/blob/1e7f5afad656c8a419cb67083dd5f4801d1b758f/pyvistaqt/plotting.py#L146

**Reply by adam-grant-hendry** [reply]
Hmmm...I wonder if that is problematic? We can always call `self.parent().update()` to reach the `QWidget.update()` method. Might we want to wrap `update()` so we can call `BasePlotter.update()`?

The two are doing two different things.

**Reply by larsoner** [reply]
It might actually be intended behavior here, but I'm not sure. [QWidget.update](https://doc.qt.io/qt-6/qwidget.html#update) schedules a repaint, which might eventually cause a rerender like what [`BasePlotter.update()`](https://github.com/pyvista/pyvista/blob/main/pyvista/plotting/plotting.py#L1788) does. I think the fact that it waits for the event loop to pick up the update/redraw request rather than being immediate is to be expected because that's how `QWidget.update` behaves, and this is all Qt-based. I think people expecting/requiring an immediate rerender (e.g., for drawing and capturing multiple frames for an animation or something) might expect they need to call `render` directly.

**Reply by adam-grant-hendry** [reply]
When upgrading our system to use `mypy`, I get the error:

```python
Definition of "update" in base class "QWidget" is incompatible with definition in base class "BasePlotter"
```

which is where this all started. We can ignore/bypass the error in several ways. I just want to be sure of which `update()` we want to call.
